src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParser.g4

```diff
@@ -861,39 +861,41 @@ description: (
 
 reference:
       (
-            PACKAGE (DOT | CLASS)* HASH? MEMBER? parameters?
-            | (DOT | CLASS)+ HASH? MEMBER? parameters?
+            PACKAGE_CLASS HASH? MEMBER? parameters?
```

Comment:
You completely remove `CLASS` from `reference`, but we still should be able to determine `CLASS` if we have `HASH`.
```
package.foo#method
```
In this case `foo` can only be a class, isn't it?